### PR TITLE
pre-commit: Replace flake8, isort, and pyupgrade with ruff

### PR DESCRIPTION
**Pull request by cclauss**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers
why we should accept this pull request. If it fixes a bug or resolves a feature
request, be sure to link to that issue.

Related to python-social-auth/social-core#768

## Types of changes

Please check the type of change your PR introduces:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Build-related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Other information

Any other information that is important to this PR such as screenshots of how
the component looks before and after the change.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,30 +9,23 @@ repos:
...
   hooks:
-  - id: isort
-    args: [--profile=black]
-- repo: https://github.com/asottile/pyupgrade
-  rev: v3.3.1
-  hooks:
-  - id: pyupgrade
-    args: [--py37-plus]
+  - id: ruff
+    args: [--format, github]  # --fix, --exit-non-zero-on-fix]
```
Comment (nijel) [inline .pre-commit-config.yaml:16]: Why not enabling autofixer? Previously we had isort enabled, and now this behavior would be gone.

Comment (cclauss) [inline .pre-commit-config.yaml:16]:
```suggestion
    args: [--format, github, --fix, --exit-non-zero-on-fix]
```

Comment (cclauss) [reply .pre-commit-config.yaml]: Ruff has 500 rules and many have fixers but maintainers may not agree to all fixes so it is better to take things one step at a time but...  Let's jump into the deep end.

Comment (cclauss) [reply .pre-commit-config.yaml]: --fixed ;-)

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**